### PR TITLE
feat(axis): log scale improvements and options

### LINKED ISSUE #916: Auto min value on log scale
**Is your feature request related to a problem? Please describe.**
When using a log scale the min value is hard coded to `1`.

https://github.com/elastic/elastic-charts/blob/f99ba1fd91a6f8a99842996d37c2d8b954e0a191/src/scales/constants.ts#L40

This causes issues in charts with a dataset with values less than `1` causing the values to not be shown.

image

https://codesandbox.io/s/happy-fog-pkd1j?file=/src/App.tsx

**Describe the solution you'd like**
EC should detect the min value from the data. Also, allow users to set the min log value via settings.

**Describe alternatives you've considered**
Setting explicit min domain value.

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/83917

**Checklist**

- [x] this request is checked against already exist requests
- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- [x] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

closes #916

The main features added in this PR are the following:
- Set base of log scales to 10, 2 or e (`Common`, `Binary` or `Natural`)
- Set custom min limit of log values, previously hardcoded to 1
- Allow domain to fit full domain of dataset, without crossing 0

### `yMinLimit`
Screen Recording 2021-02-04 at 01 16 PM

### No `yMinLimit` set, uses full domain
Screen Recording 2021-02-04 at 01 17 PM

### `yLogBase`
Screen Recording 2021-02-04 at 01 18 PM

### Checklist
- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

File: api/charts.api.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import { $Values } from 'utility-types';
 import { ComponentType } from 'react';
-import React from 'react';
+import { default as React_2 } from 'react';
```
Comment (nickofthyme) [inline api/charts.api.md:9]: @markov00 I was able to fix [this](https://github.com/elastic/elastic-charts/pull/1014/checks?check_run_id=1857137839) error by upgrading the `api-extractor` but it made these changes. Any ideas to `React`?

Comment (markov00) [reply api/charts.api.md:9]: This usually happens when you are using JSX without importing React or when importing react as `import * from React` but I've checked your PR and it doesn't seem to touch anything like that. I will investigate on it a bit

File: api/charts.api.md [hunk 2]
```diff
@@ -1923,10 +1952,12 @@ export interface SettingsSpec extends Spec {
...
     tooltip: TooltipSettings;
-    // Warning: (ae-forgotten-export) The symbol "Domain" needs to be exported by the entry point index.d.ts
-    //
     // (undocumented)
-    xDomain?: Domain | DomainRange;
+    xDomain?: ContinuousDomain | OrdinalDomain | DomainRange;
+    xLogBase: LogBase;
+    xLogMinLimit?: number;
+    yLogBase: LogBase;
+    yLogMinLimit?: number;
```
[the remark this answers is not in the document]
Comment (nickofthyme) [reply api/charts.api.md:1960]: OK see a26fd2c

**Review by markov00 — COMMENTED** [review]
Changes LGTM

**Pull request merged**